prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -217,64 +218,70 @@ class ReportFunction {
         workflowEngine.db.transact { txn ->
             workflowEngine
                 .translator
-                .filterAndTranslateByService(validatedRequest.report!!, validatedRequest.defaults)
-                .forEach { (report, service) ->
+                .filterAndTranslateByReceiver(validatedRequest.report!!, validatedRequest.defaults)
+                .forEach { (report, receiver) ->
+                    val organization = workflowEngine.metadata.findOrganization(receiver.organizationName)!!
                     sendToDestination(
-                        report, service, context, workflowEngine,
-                        validatedRequest, destinations, actionHistory, txn
+                        report,
+                        organization,
+                        receiver,
+                        context,
+                        workflowEngine,
+                        validatedRequest,
+                        destinations,
+                        actionHistory,
+                        txn
                     )
                 }
         }
     }
 
     private fun sendToDestination(
         report: Report,
-        service: OrganizationService,
+        organization: Organization,
+        receiver: Receiver,
         context: ExecutionContext,
         workflowEngine: WorkflowEngine,
         validatedRequest: ValidatedRequest,
         destinations: MutableList<String>,
         actionHistory: ActionHistory,
         txn: DataAccessTransaction
     ) {
-        val serviceDescription = if (service.organization.services.size > 1)
-            "${service.organization.description} (${service.name})"
-        else
-            service.organization.description
+        val receiverDescription = "${organization.description} (${receiver.name})"
```

Comment:
Counting the number of receivers associated with an organization is another query when a DB is the backing store. Didn't seem worth it here.